### ISSUE TITLE
es6 shorthand for class methods fails func-names and space-before-function-parenteses

### ISSUE THREAD

**Issue by joeybaker** [eslint/eslint#2350]
Given the following code:

```js
class a {
  method () {
  }
}
```

and the following eslintrc:

```json
"rules": {
    "func-names": 1,
    "space-before-function-parentheses": [
      1,
      "always"
    ]
}
```

eslint throws a warning for each rule, when it should not.

```
     2: Missing function expression name. (func-names)
     2: Missing space before function parentheses. (space-before-function-parentheses)
```

**Comment by nzakas** [comment]
Can you provide the actual output?

**Comment by joeybaker** [comment]
Yup. Initial comment updated.

**Comment by btmills** [comment]
`space-before-function-parentheses` is deprecated in favor of `space-before-function-paren`. I believe the latter has been uupdated for class methods. Can you try that? It should fix the second error.

**Comment by joeybaker** [comment]
@btmills – bingo. Thanks. Is there a way we can show deprecation messages when warnings are printed for deprecated rules?

**Comment by ilyavolodin** [comment]
Deprecated rules will be removed with the next major release. Currently, only documentation is marked with deprecated tag, rules themselves don't know anything. Because of that, adding a warning will be problematic.
